ipatests/test_integration/test_sssd.py

```diff
@@ -286,3 +287,45 @@ def get_cache_update_time(obj_kind, obj_name):
         finally:
             sssd_conf_backup.restore()
             tasks.clear_sssd_cache(self.master)
+
+    def test_ext_grp_with_ldap(self):
+        """User and group with same name should not break reading AD user data.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4073
+
+        When aduser is added in extrnal group and this group is added
+        in group with same name of nonprivate ipa user and possix id, then
+        lookup of aduser and group should be successful when cache is empty.
+        """
+        cmd = self.master.run_command(['sssd', '--version'])
+        sssd_version = platform_tasks.parse_ipa_version(
+            cmd.stdout_text.strip())
+        if sssd_version <= platform_tasks.parse_ipa_version('2.2.2'):
+            pytest.xfail("Fix for https://pagure.io/SSSD/sssd/issue/4073 "
```

Comment:
Let me explain the difference between skip and xfail:

>  A **skip** means that you expect your test to pass only if some conditions are met, otherwise pytest should skip running the test altogether. [...]
A **xfail** means that you expect a test to fail for some reason. A common example is a test for a feature not yet implemented, or a bug not yet fixed. When a test passes despite being expected to fail (marked with pytest.mark.xfail), it’s an xpass and will be reported in the test summary.
-- http://doc.pytest.org/en/latest/skipping.html

We use xfails to to double check ourselves that we properly understand the conditions that must be met for test to pass. Here we could use it to verify that we have properly identified minimum sssd version which has the bug fixed.

But the way it is written now, the test has no chance to succeed and to be eventually reported as "XPASS".

I suggest to do one of the following:
* Replace xfail with skip.
* properly use xfail inverting success/fail of the assertion.
* wait until #3924 is merged and use the xfail context manager introduced in that PR

Reply:
> But the way it is written now, the test has no chance to succeed with older version of sssd and to be eventually reported as "XPASS".
with older version i am getting xfail
http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/49484806-64fc-11ea-a229-fa163e97fbb7/report.html
which is expected to fail there with 2.2.2

Reply:
"xfail" means we are making additional check for correctness of condition. Imagine, the fix is somehow added to the 2.2.2. Or maybe the assumption that 2.2.3 is needed was wrong and 2.2.2 was working from the beginning. With the current implementation we will always have "expected failure" with 2.2.2 even if it  does not contain the bug (or the bug is fixed in the future). The proper implementation should produce "unexpected success" in that case so that we can review the test and  adapt the condition. That current behavior is exactly what "skip" does so calling it "xfails" is misleading.

> with older version i am getting xfail

The problem is that you will always get xfail with older version, even if this older version is fixed, which is not the expected behavior of "xfail". If you do not care about it, please use "skip" to avoid misleading. Otherwise, xfail should invert the logic of the assertion.